src/hooks/useCreatePlayer.ts

```diff
@@ -0,0 +1,30 @@
+import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
+import axios from "axios";
+import { useEffect } from "react";
+import { API_URL } from "../settings";
+import { setPlayerName } from "../state/playerSlice";
+import { joinGameRoomAsHostAction } from "../state/socketActions/joinGameRoomAction";
+
+export function useCreatePlayer() {
+    const { name } = useAppSelector(state => state.player);
+    const { code } = useAppSelector(state => state.game);
+    const dispatch = useAppDispatch();
+    useEffect(() => {
+        if (!name || !code) return;
+        axios
+            .post(`${API_URL}/players`,
+                {
+                    "name": name,
+                    "joinCode": code
+                })
+            .then(response => {
+                // Not sure what we should do with the response
```

Comment:
My suggestion would be to add a field to the GameSlice like `hasJoinedGame`, and then set it to true when this post request is successful. We can then use the `hasJoinedGame` field to conditionally render stuff later on

Reply:
Great idea!